atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/BucketBasedTargetedSweeper.java

```diff
@@ -239,12 +242,13 @@ private static DefaultBucketAssigner createBucketAssigner(
             DefaultSweepAssignedBucketStore sweepAssignedBucketStore,
             List<SweeperStrategy> strategies,
             int numShards,
+            Supplier<Long> maxCoarsePartitionsPerBucketForNonPuncherClose,
```

Comment:
nit: we should take Refreshable I think, unless you expect there to be situations where this doesn't come from config? For consistency, and for stricter type-checking.